File: sdk/src/main/java/com/meniga/sdk/webservices/requests/GetChallenge.java
package com.meniga.sdk.webservices.requests;

import java.util.UUID;

/**
 * Copyright 2017 Meniga Iceland Inc.
 */
public class GetChallenge extends QueryRequestObject {

	public UUID id;

	@Override
	public long getValueHash() {
		return id != null ? id.hashCode() : 0;
	}
}

File: sdk/src/main/java/com/meniga/sdk/webservices/challenge/GetChallenge.kt
/*
 * Copyright 2017-2018 Meniga Iceland Inc.
 */
package com.meniga.sdk.webservices.challenge

import com.meniga.sdk.webservices.requests.QueryRequestObject

import java.util.UUID

data class GetChallenge(
        @JvmField var id: UUID
) : QueryRequestObject() {

    override fun getValueHash(): Long = hashCode().toLong()
}
